crates/orchestrator/src/client.rs

```diff
@@ -185,30 +185,35 @@ impl OrchestratorClient {
         OrchestratorClient { client, identity }
     }
 
-    /// Sends an identify message to the orchestrator and attempts to get its config
-    /// Returns both the `node_index` and the run configuration without peer's public config from the orchestrator
-    /// Will block until both are returned
+    /// Get the config from the orchestrator.
+    /// If the identity is provided, register the identity with the orchestrator.
+    /// If not, just retrieving the config (for passive observers)
+    ///
     /// # Panics
     /// if unable to convert the node index from usize into u64
     /// (only applicable on 32 bit systems)
     #[allow(clippy::type_complexity)]
-    pub async fn get_config_without_peer<K: SignatureKey, E: ElectionConfig>(
+    pub async fn get_config<K: SignatureKey, E: ElectionConfig>(
```

Comment:
All the others LGTM. The only thing is I'm more inclined to keep these comments (maybe together with the name `get_config_without_peer`) to emphasize this step haven't collected peers' info yet, so that it will not cause confusion with another two functions `get_config_after_peer_collected` and `get_config` in `trait RunDA`.

Reply:
Ahh okay, that makes sense. I will change it. Question: are the public keys of the peers available at this step? It definitely works for the whitelist during testing, but just curious.

Reply:
The node's own public keys are available, but its peers' public keys are not collected by it at this step yet.

Reply:
Do you think we can unify the two functions? AKA change it so we don't return the initial config until everyone has posted their public keys? Without the need for the second function?

It looks like we already have similar logic: https://github.com/EspressoSystems/HotShot/blob/22631060ef5e4fd9ce97e3ee843cb0fa8856e09e/crates/orchestrator/src/lib.rs#L230-L235


WDYT?

Reply:
Sorry I just realized `get_config_without_peer` is updated to allow for pulling the list of keys (although not sure how, looks like the function just add one more rounds of check `if let Some(identity) = identity`, would appreciate any more info on this). The function `get_complete_config` will only return the config until everyone has posted their public keys, where its first step uses `get_config_without_peer` to load configs from file, and the second step uses `post_and_wait_all_public_keys` to gather peers' public keys.

If it's true `get_config_without_peer` already pulled the list of keys and I could understand how, then the second step is not needed anymore.

Reply:
Okay this makes sense, so instead I should get the public keys with `get_config_after_peer_collected`. It was probably just a timing thing that allowed it to work for `before_collected`

Reply:
Updated !